Test failure on macos 15.2 (Sequoia): certificate mismatch in t/local/32_x509_get_cert_info.t

```
cpanm (App::cpanminus) 1.7048 on perl 5.041007 built for darwin-2level
Work directory is /Volumes/chambord/Users/ether/.cpanm/work/1737241069.21685
You have make /usr/bin/make
You have LWP 6.77
You have /usr/bin/tar: bsdtar 3.5.3 - libarchive 3.5.3 zlib/1.2.12 liblzma/5.4.3 bz2lib/1.0.8 
You have /usr/bin/unzip
Searching Net::SSLeay () on cpanmetadb ...
--> Working on Net::SSLeay
Fetching http://www.cpan.org/authors/id/C/CH/CHRISN/Net-SSLeay-1.94.tar.gz
-> OK
Unpacking Net-SSLeay-1.94.tar.gz
Entering Net-SSLeay-1.94
Checking configure dependencies from META.json
Checking if you have constant 0 ... Yes (1.33)
Checking if you have File::Spec::Functions 0 ... Yes (3.93)
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.70)
Checking if you have English 0 ... Yes (1.11)
Checking if you have Text::Wrap 0 ... Yes (2024.001)
Configuring Net-SSLeay-1.94
Running Makefile.PL
Do you want to run external tests?
These tests *will* *fail* if you do not have network connectivity. [n] n
*** If there are build errors, test failures or run-time malfunctions,
    try to use the same compiler and options to compile your OpenSSL,
    Perl, and Net::SSLeay.
*** Found OpenSSL-3.4.0 installed in /opt/local
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Net::SSLeay
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have English 0 ... Yes (1.11)
Checking if you have SelectSaver 0 ... Yes (1.02)
Checking if you have Cwd 0 ... Yes (3.92)
Checking if you have Storable 0 ... Yes (3.35)
Checking if you have Socket 0 ... Yes (2.038)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.70)
Checking if you have Test::More 0.60_01 ... Yes (1.302207)
Checking if you have File::Basename 0 ... Yes (2.86)
Checking if you have base 0 ... Yes (2.27)
Checking if you have Config 0 ... Yes (5.041007)
Checking if you have MIME::Base64 0 ... Yes (3.16_01)
Checking if you have Test::Builder 0 ... Yes (1.302207)
Checking if you have File::Spec::Functions 0 ... Yes (3.93)
Checking if you have Carp 0 ... Yes (1.54)
Checking if you have Scalar::Util 0 ... Yes (1.68_01)
Building and testing Net-SSLeay-1.94
AutoSplitting blib/lib/Net/SSLeay.pm (blib/lib/auto/Net/SSLeay)
blib/lib/Net/SSLeay.pm: some names are not unique when truncated to 8 characters:
 directory blib/lib/auto/Net/SSLeay:
  do_https3.al, do_https2.al, do_https4.al, do_https.al truncate to do_https
  do_httpx3.al, do_httpx2.al, do_httpx4.al truncate to do_httpx
  get_https.al, get_https3.al, get_https4.al, get_http.al, get_http3.al, get_http4.al, get_httpx.al, get_httpx3.al, get_httpx4.al truncate to get_http
  head_https.al, head_https3.al, head_https4.al, head_http.al, head_http3.al, head_http4.al, head_httpx.al, head_httpx3.al, head_httpx4.al truncate to head_htt
  post_https.al, post_https3.al, post_https4.al, post_http.al, post_http3.al, post_http4.al, post_httpx.al, post_httpx3.al, post_httpx4.al truncate to post_htt
  put_https.al, put_https3.al, put_https4.al, put_http.al, put_http3.al, put_http4.al, put_httpx.al, put_httpx3.al, put_httpx4.al truncate to put_http
  ssl_read_all.al, ssl_read_until.al, ssl_read_CRLF.al truncate to ssl_read
  ssl_write_all.al, ssl_write_CRLF.al truncate to ssl_writ
  tcp_read_all.al, tcp_read_until.al, tcp_read_CRLF.al truncate to tcp_read
  tcp_write_all.al, tcp_write_CRLF.al truncate to tcp_writ
Running Mkbootstrap for SSLeay ()
chmod 644 "SSLeay.bs"
"/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/bin/perl5.41.7" -MExtUtils::Command::MM -e 'cp_nonempty' -- SSLeay.bs blib/arch/auto/Net/SSLeay/SSLeay.bs 644
"/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/bin/perl5.41.7" "/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/lib/5.41.7/ExtUtils/xsubpp"  -typemap '/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/lib/5.41.7/ExtUtils/typemap' -typemap '/Volumes/chambord/Users/ether/.cpanm/work/1737241069.21685/Net-SSLeay-1.94/typemap'  SSLeay.xs > SSLeay.xsc
mv SSLeay.xsc SSLeay.c
cc -c  -I"/opt/local/include" -DOPENSSL_API_COMPAT=908 -DNET_SSLEAY_PERL_VERSION=5041007 -fno-common -DPERL_DARWIN -mmacosx-version-min=15.1 -DNO_THREAD_SAFE_QUERYLOCALE -DNO_POSIX_2008_LOCALE -DHAS_BROKEN_LANGINFO_CODESET -fno-strict-aliasing -pipe -fstack-protector-strong -I/opt/local/include -Wno-error=implicit-function-declaration -O3   -DVERSION=\"1.94\" -DXS_VERSION=\"1.94\"  "-I/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/lib/5.41.7/darwin-2level/CORE"   SSLeay.c
rm -f blib/arch/auto/Net/SSLeay/SSLeay.bundle
LD_RUN_PATH="/opt/local/lib" cc -Wl,-rpath,"/opt/local/lib" -mmacosx-version-min=15.1 -bundle -undefined dynamic_lookup -L/opt/local/lib -L/opt/local -L/opt/local/lib -fstack-protector-strong  SSLeay.o  -o blib/arch/auto/Net/SSLeay/SSLeay.bundle  \
	   -L/opt/local/lib -L/opt/local -lssl -lcrypto -lz   \
	  
chmod 755 blib/arch/auto/Net/SSLeay/SSLeay.bundle
"/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/bin/perl5.41.7" -MExtUtils::Command::MM -e 'cp_nonempty' -- SSLeay.bs blib/arch/auto/Net/SSLeay/SSLeay.bs 644
PERL_DL_NONLAZY=1 "/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/bin/perl5.41.7" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/local/*.t t/handle/local/*.t
t/handle/local/05_use.t ..................... ok
t/local/02_pod_coverage.t ................... skipped: These tests are for only for release candidate testing. Enable with RELEASE_TESTING=1
# 
# Testing Net::SSLeay 1.94
# 
# Perl information:
#   Version:         '5.041007'
#   Executable path: '/Volumes/chambord/Users/ether/perl5/perlbrew/perls/41.7/bin/perl5.41.7'
# 
# Library version with OpenSSL_version_num():
#   OPENSSL_VERSION_NUMBER: '0x30400000'
# 
# Library information with SSLeay_version() and OpenSSL_version():
#   SSLEAY_VERSION:              'OpenSSL 3.4.0 22 Oct 2024'
#   SSLEAY_CFLAGS:               'compiler: /usr/bin/clang -fPIC -arch arm64 -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -DL_ENDIAN -DOPENSSL_PIC -D_REENTRANT -DOPENSSL_BUILDING_OPENSSL -DZLIB -DNDEBUG -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk'
#   SSLEAY_BUILT_ON:             'built on: Tue Dec  3 14:33:57 2024 UTC'
#   SSLEAY_PLATFORM:             'platform: darwin64-arm64-cc'
#   SSLEAY_DIR:                  'OPENSSLDIR: "/opt/local/libexec/openssl3/etc/openssl"'
#   OPENSSL_ENGINES_DIR:         'ENGINESDIR: "/opt/local/libexec/openssl3/lib/engines-3"'
#   OPENSSL_MODULES_DIR:         'MODULESDIR: "/opt/local/libexec/openssl3/lib/ossl-modules"'
#   OPENSSL_CPU_INFO:            'CPUINFO: OPENSSL_armcap=0x987d'
#   OPENSSL_VERSION_STRING:      '3.4.0'
#   OPENSSL_FULL_VERSION_STRING: '3.4.0'
# 
# Library version information with OPENSSL_version_*():
#   OPENSSL_version_major():          '3'
#   OPENSSL_version_minor():          '4'
#   OPENSSL_version_patch():          '0'
#   OPENSSL_version_pre_release():    ''
#   OPENSSL_version_build_metadata(): ''
# 
# Library information with OPENSSL_info():
#   OPENSSL_INFO_CONFIG_DIR:             '/opt/local/libexec/openssl3/etc/openssl'
#   OPENSSL_INFO_ENGINES_DIR:            '/opt/local/libexec/openssl3/lib/engines-3'
#   OPENSSL_INFO_MODULES_DIR:            '/opt/local/libexec/openssl3/lib/ossl-modules'
#   OPENSSL_INFO_DSO_EXTENSION:          '.dylib'
#   OPENSSL_INFO_DIR_FILENAME_SEPARATOR: '/'
#   OPENSSL_INFO_LIST_SEPARATOR:         ':'
#   OPENSSL_INFO_SEED_SOURCE:            'os-specific'
#   OPENSSL_INFO_CPU_SETTINGS:           'OPENSSL_armcap=0x987d'
t/local/03_use.t ............................ ok
t/local/04_basic.t .......................... ok
t/local/05_passwd_cb.t ...................... ok
t/local/01_pod.t ............................ ok
t/local/06_tcpecho.t ........................ ok
t/local/09_ctx_new.t ........................ ok
t/local/08_pipe.t ........................... ok
t/local/10_rand.t ........................... ok
t/local/07_sslecho.t ........................ ok
t/local/15_bio.t ............................ ok
t/local/20_functions.t ...................... ok
t/local/11_read.t ........................... ok
t/local/22_provider_try_load.t .............. ok
t/local/22_provider.t ....................... ok
t/local/21_constants.t ...................... ok
t/local/22_provider_try_load_zero_retain.t .. ok
t/local/30_error.t .......................... ok
t/local/23_openssl_init.t ................... ok
t/local/34_x509_crl.t ....................... ok
t/local/35_ephemeral.t ...................... skipped: LibreSSL and OpenSSL 1.1.0 removed support for ephemeral/temporary RSA private keys

#   Failed test 'X509V3_EXT_print nid=86	extended-cert.cert.pem:6'
#   at t/local/32_x509_get_cert_info.t line 273.
#          got: 'email:intermediate-ca@net-ssleay.example, URI:http://intermediate-ca.net-ssleay.example, DNS:intermediate-ca.net-ssleay.example, Registered ID:1.2.0.0, IP Address:192.168.0.1, IP Address:FD25:F814:AFB5:9873:0:0:0:1, othername: emailAddress:ica@net-ssleay.example'
#     expected: 'email:intermediate-ca@net-ssleay.example, URI:http://intermediate-ca.net-ssleay.example, DNS:intermediate-ca.net-ssleay.example, Registered ID:1.2.0.0, IP Address:192.168.0.1, IP Address:FD25:F814:AFB5:9873:0:0:0:1, othername: emailAddress::ica@net-ssleay.example'

#   Failed test 'X509V3_EXT_print nid=85	extended-cert.cert.pem:8'
#   at t/local/32_x509_get_cert_info.t line 273.
#          got: 'email:john.doe@net-ssleay.example, URI:http://johndoe.net-ssleay.example, DNS:johndoe.net-ssleay.example, Registered ID:1.2.3.4, IP Address:192.168.0.2, IP Address:FD25:F814:AFB5:9873:0:0:0:2, othername: emailAddress:jd@net-ssleay.example'
#     expected: 'email:john.doe@net-ssleay.example, URI:http://johndoe.net-ssleay.example, DNS:johndoe.net-ssleay.example, Registered ID:1.2.3.4, IP Address:192.168.0.2, IP Address:FD25:F814:AFB5:9873:0:0:0:2, othername: emailAddress::jd@net-ssleay.example'
# Looks like you failed 2 tests of 746.
t/local/32_x509_get_cert_info.t ............. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/746 subtests 
t/local/38_priv-key.t ....................... ok
t/local/37_asn1_time.t ...................... ok
t/local/36_verify.t ......................... ok
t/local/39_pkcs12.t ......................... ok
t/local/40_npn_support.t .................... ok
t/local/42_info_callback.t .................. ok
t/local/41_alpn_support.t ................... ok
t/local/31_rsa_generate_key.t ............... ok
t/local/43_misc_functions.t ................. ok
t/local/45_exporter.t ....................... ok
t/local/46_msg_callback.t ................... ok
t/local/44_sess.t ........................... ok
t/local/47_keylog.t ......................... ok
t/local/33_x509_create_cert.t ............... ok
t/local/61_threads-cb-crash.t ............... skipped: Threads not supported on this system
t/local/62_threads-ctx_new-deadlock.t ....... skipped: Threads not supported on this system
t/local/48_client_hello_callback.t .......... ok
t/local/50_digest.t ......................... ok
t/local/63_ec_key_generate_key.t ............ ok
t/local/65_security_level.t ................. ok
t/local/64_ticket_sharing.t ................. ok
t/local/kwalitee.t .......................... skipped: These tests are for only for release candidate testing. Enable with RELEASE_TESTING=1
t/local/65_ticket_sharing_2.t ............... ok
t/local/66_curves.t ......................... ok

Test Summary Report
-------------------
t/local/32_x509_get_cert_info.t           (Wstat: 512 (exited 2) Tests: 746 Failed: 2)
  Failed tests:  633, 651
  Non-zero exit status: 2
Files=47, Tests=2787,  0 wallclock secs ( 0.14 usr  0.06 sys +  2.25 cusr  0.47 csys =  2.92 CPU)
Result: FAIL
Failed 1/47 test programs. 2/2787 subtests failed.
make: *** [test_dynamic] Error 255
-> FAIL Installing Net::SSLeay failed. See /Volumes/chambord/Users/ether/.cpanm/work/1737241069.21685/build.log for details. Retry with --force to force install it.

```

--- a/Changes
+++ b/Changes
@@ -7,6 +7,9 @@ Revision history for Perl extension Net::SSLeay.
 	- Update GitHub Actions CI workflow. A number of test jobs
 	  were broken because some GitHub runners were discontinued,
 	  changes in QEMU setup, changes in Cygwin, etc.
+	- Adjust test 32_x509_get_cert_info.t to match formatting
+	  changes in OpenSSL 3.4.0 and 3.4.1. Thanks to Sebastian
+	  Andrzej Siewior for the patches.
 
 1.94 2024-01-08
 	- New stable release incorporating all changes from developer releases 1.93_01

--- a/t/local/32_x509_get_cert_info.t
+++ b/t/local/32_x509_get_cert_info.t
@@ -188,6 +188,10 @@ for my $f (keys (%$dump)) {
                   ) {
                       $ext_data =~ s{(othername:) [^, ]+}{$1<unsupported>}g;
                   }
+                  # Starting with 3.4.0 the double colon in emailAddress has been removed.
+                  if (Net::SSLeay::SSLeay >= 0x30400000) {
+                      $ext_data =~ s{emailAddress::}{emailAddress:};
+                  }
               }
               elsif ( $nid == 89 ) {
                   # The output formatting for certificate policies has a

--- a/t/local/32_x509_get_cert_info.t
+++ b/t/local/32_x509_get_cert_info.t
@@ -218,6 +218,9 @@ for my $f (keys (%$dump)) {
                       # OpenSSL 1.0.0 to 1.1.1:
                       $ext_data =~ s{(Full Name:\n  )}{\n$1}g;
                       $ext_data .= "\n";
+                  } elsif ( Net::SSLeay::SSLeay >  0x3040000f ) {
+                      $ext_data =~ s{(\nFull Name:)}{\n$1}g;
+                      $ext_data .= "\n";
                   }
               }
               elsif ( $nid == 126 ) {

